champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java

```diff
@@ -23,19 +23,19 @@
 import org.bukkit.event.block.Action;
 import org.bukkit.event.entity.EntityDamageEvent;
 import org.bukkit.event.player.PlayerInteractEvent;
-import org.bukkit.inventory.EquipmentSlot;
 import org.bukkit.potion.PotionEffect;
 import org.bukkit.potion.PotionEffectType;
 
-import java.util.HashSet;
-import java.util.Set;
+import java.util.Iterator;
+import java.util.Map;
 import java.util.UUID;
+import java.util.WeakHashMap;
 
 @Singleton
 @BPvPListener
 public class Agility extends Skill implements InteractSkill, CooldownSkill, Listener {
 
-    private final Set<UUID> active = new HashSet<>();
+    private final WeakHashMap<UUID, Long> active = new WeakHashMap<>();
```

Comment:
No need to use a WeakHashMap if storing a UUID, just do a regular HashMap

Reply:
Changed